### PR TITLE
[DEV-1729] Support cross aggregation in aggregate_asat

### PR DESCRIPTION
**Pull request by YS-L**
## Description

This adds support for cross aggregation (specified via the `category` groupby parameter) in `aggregate_asat`.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/sql/specs.py [hunk 1]
```diff
@@ -288,6 +289,26 @@ def get_aggregation_source(
...
+            Groupby parameters
+
+        Returns
+        -------
+        str
+        """
+        args = [parameters.agg_func, parameters.parent]
+        args.extend(parameters.keys or [None])  # type: ignore[list-item]
+        args.append(parameters.value_by or None)
+        return self.construct_agg_result_name(*args)
```
Comment (YS-L) [inline featurebyte/query_graph/sql/specs.py:310]: This is to handle the case when the same node is used by multiple group by operations (e.g. with different keys).

Comment (gee-senbong) [reply featurebyte/query_graph/sql/specs.py:310]: nit: Maybe we could use something like
```
args = [
    parameters.agg_func,
    parameters.parent,
    *(parameters.keys or [None]),
    parameters.value_by or None,
]
```

File: tests/integration/api/test_feature_correctness.py [hunk 2]
```diff
@@ -104,9 +115,10 @@ def _extract_current_record(sub_df):
     df_current = df.groupby(natural_key_column_name).apply(_extract_current_record)
 
     df_filtered = df_current[df_current[entity_column_name] == entity_value]
-    out = agg_func(df_filtered[variable_column_name])
 
-    return out
+    return apply_agg_func_on_filtered_dataframe(
+        agg_func, category, df_filtered, variable_column_name
+    )
```
Comment (gee-senbong) [inline tests/integration/api/test_feature_correctness.py:121]: 👍

**Review by gee-senbong — APPROVED** [review]
LGTM

**Pull request merged**